app/controllers/lifehacks_controller.rb

```diff
@@ -32,11 +32,35 @@ def create
   end
 
   def search
-    if params[:search].nil?
-      @lifehacks = []
-    else
-      @lifehacks = Lifehack.where("title ILIKE ?", "%#{params[:search]}%")
+    lifehacks = []
+    if search_params[:all] && search_params[:all] != ""
+      lifehacks << Lifehack.where(
+        "title ILIKE ? OR description ILIKE ?",
+        "%#{search_params[:all]}%", "%#{search_params[:all]}%"
+      )
+    end
+    if search_params[:title] && search_params[:title] != ""
+      lifehacks << Lifehack.where("title ILIKE ?", "%#{search_params[:title]}%")
+    end
+    if search_params[:description] && search_params[:description] != ""
+      lifehacks << Lifehack.where(
+        "description ILIKE ?", "%#{search_params[:description]}%"
+      )
     end
+    if search_params[:user] && search_params[:user] != ""
+      user_details = search_params[:user]
+      sql_query = "first_name ILIKE ? OR last_name ILIKE ?"
+      if user_details.include?(" ")
+        user_details = user_details.split(" ")
+        user = User.where(
+          sql_query, "%#{user_details[0]}%", "%#{user_details[1]}%"
+        )
+      else
+        user = User.where(sql_query, "%#{user_details}%", "%#{user_details}%")
+      end
+      lifehacks << Lifehack.where(creator: user)
+    end
+    @lifehacks = lifehacks.flatten.uniq
   end
```

Comment:
This method could use a refactor. There are a bunch of active record calls happening here which can be abstracted to where we interact with the DB which is on the model. Since we will want to call the method on the class itself you'll want to use class methods... something like: 

```
def self.search(query)
#active record query so for example
where("title ILKE ?", "%#{search_params[:all]%")
end
```

Then in the controller you can call 
```
if search_params[:all] != ""
      @lifehacks = Lifehack.search(search_params[:all])
# then you can add any other models you searching
```
This will allow you to write search class method for any models you want to search and they can all use the same search template and which you could put on the homes controller. Let me know if you have questions and want to talk about this more.